### DISCUSSION TITLE
How to modify required online script of a script?

### DISCUSSION THREAD

**Discussion by byzod** [violentmonkey/violentmonkey#1837] [Q&A]
Platform: firefox 115b6 x64, window 10

**The concept**: I can edit `script A` locally, but if the script I want to edit is a base script `@require` by another `script B`, how can I edit `script B` by edit its required **online** resource `script A`? I think requiring a local resource is not likely possible for firefox's strict security control

**The practical problem**: the zoom hotkey of [manga loader](https://greasyfork.org/en/scripts/692-manga-loader) is no longer working on firefox because it use a [deprecated method](http://udn.realityripple.com/docs/Web/API/InstallTrigger) to detect firefox:

    if(typeof InstallTrigger !== 'undefined') {
        UI.keys.ZOOM_IN = 61;
        UI.keys.ZOOM_OUT = 173;
        UI.keys.RESET_ZOOM = 48;
    }

Then I modify it to make it work, however, the Manga Loader NSFW use the base script by require it from greasyfork repo, my edition in local can't affect it
    
    // ==UserScript==
    // @name       Manga Loader NSFW
    // @namespace  http://www.fuzetsu.com/MangaLoaderNSFW
    ...
    // @require https://greasyfork.org/scripts/692-manga-loader/code/Manga%20Loader.user.js?29
    // ==/UserScript==

Is there any reliable way to fix Manga Loader NSFW

**Comment by tophf** [comment]
Try asking the author of the script. Otherwise, if the script is abandoned, you can edit it locally. If it's inside a `@require` then you'll need to remove the entire `@require` line in the main script and paste the contents of the edited script afterwards.
